tst/com/amazon/corretto/crypto/provider/test/EvpKeyAgreementSpecificTest.java

```diff
@@ -32,27 +32,25 @@
     private static final Class<?> SPI_CLASS;
     private static final int EC_TYPE = 408;
     private static final int DH_TYPE = 28;
-    private final KeyPair EC_KEYPAIR;
-    private final KeyPair DH_KEYPAIR;
+    private static final KeyPair EC_KEYPAIR;
+    private static final KeyPair DH_KEYPAIR;
 
     static {
       try {
           SPI_CLASS = Class.forName("com.amazon.corretto.crypto.provider.EvpKeyAgreement");
-      } catch (final ClassNotFoundException ex) {
-          throw new AssertionError(ex);
-      }
-    }
 
-    public EvpKeyAgreementSpecificTest() throws GeneralSecurityException {
-        // Force loading of native library
-        KeyAgreement.getInstance("ECDH", AmazonCorrettoCryptoProvider.INSTANCE);
+          // Force loading of native library
+          KeyAgreement.getInstance("ECDH", AmazonCorrettoCryptoProvider.INSTANCE);
 
-        KeyPairGenerator gen = KeyPairGenerator.getInstance("EC");
-        gen.initialize(new ECGenParameterSpec("NIST P-224"));
-        EC_KEYPAIR = gen.generateKeyPair();
-        gen = KeyPairGenerator.getInstance("DH");
-        gen.initialize(1024);
-        DH_KEYPAIR = gen.generateKeyPair();
+          KeyPairGenerator gen = KeyPairGenerator.getInstance("EC");
+          gen.initialize(new ECGenParameterSpec("NIST P-224"));
+          EC_KEYPAIR = gen.generateKeyPair();
+          gen = KeyPairGenerator.getInstance("DH");
+          gen.initialize(1024);
+          DH_KEYPAIR = gen.generateKeyPair();
+      } catch (final Exception ex) {
```

Comment:
Why are we putting all of this into a generic try/catch?

Reply:
It's the static initializer for a test case. If this fails in any way, we should just abort. There is no value to a more specific try-catch block since all exceptions result in an "AssertionError" in the end anyway.

Reply:
ah, got it. I see now that it's all part of the static initializer now. :+1: